internal/cmd/agent/deployer/monitor/updatestatus.go

```diff
@@ -258,6 +258,7 @@ func calculateResourceCounts(all []fleet.BundleDeploymentResource, nonReady []fl
 	for _, r := range modified {
 		if r.Create {
 			counts.Missing++
+			counts.DesiredReady++ // plan.Objects do not include missing resources
```

Comment:
I'd be tempted to expand on this comment a bit, to make it easier to understand for future readers:
```suggestion
                         // counts are determined from `all`, itself computed from `plan.Objects` (see `toBundleDeploymentResources` which do not include missing resources
			counts.DesiredReady++
```

... which makes me wonder: would it not make more sense for this correction to happen in `toBundleDeploymentResources`, which could take `modifiedResources` as an additional argument to add resources to be created to the total?

Reply:
I've refactored it to a separate function, hopefully making it clearer